src/newTest/java/org/opensearch/test/framework/cluster/LocalCluster.java

```diff
@@ -0,0 +1,373 @@
+
+
+public class LocalCluster extends ExternalResource implements AutoCloseable, OpenSearchClientProvider {
+
+    private static final Logger log = LogManager.getLogger(LocalCluster.class);
+
+    static {
+        System.setProperty("security.default_init.dir", new File("./securityconfig").getAbsolutePath());
+    }
+
+    protected static final AtomicLong num = new AtomicLong();
+
+    private final List<Class<? extends Plugin>> plugins;
+    private final ClusterConfiguration clusterConfiguration;
+    private final TestSecurityConfig testSecurityConfig;
+    private Settings nodeOverride;
+    private final String clusterName;
+    private final MinimumSecuritySettingsSupplierFactory minimumOpenSearchSettingsSupplierFactory;
+    private final TestCertificates testCertificates;
+    private final List<LocalCluster> clusterDependencies;
+    private final Map<String, LocalCluster> remotes;
+    private volatile LocalOpenSearchCluster localOpenSearchCluster;
+	private final List<TestIndex> testIndices;
+
+    private LocalCluster(String clusterName, TestSecurityConfig testSgConfig, Settings nodeOverride,
+            ClusterConfiguration clusterConfiguration, List<Class<? extends Plugin>> plugins, TestCertificates testCertificates,
+            List<LocalCluster> clusterDependencies, Map<String, LocalCluster> remotes, List<TestIndex> testIndices) {
+        this.plugins = plugins;
+        this.testCertificates = testCertificates;
+        this.clusterConfiguration = clusterConfiguration;
+        this.testSecurityConfig = testSgConfig;
+        this.nodeOverride = nodeOverride;
+        this.clusterName = clusterName;
+        this.minimumOpenSearchSettingsSupplierFactory = new MinimumSecuritySettingsSupplierFactory(testCertificates);
+        this.remotes = remotes;
+        this.clusterDependencies = clusterDependencies;
+        this.testIndices = testIndices;
+    }
+
+    @Override
+    public void before() throws Throwable {
+        if (localOpenSearchCluster == null) {
+            for (LocalCluster dependency : clusterDependencies) {
+                if (!dependency.isStarted()) {
+                    dependency.before();
+                }
+            }
+
+            for (Map.Entry<String, LocalCluster> entry : remotes.entrySet()) {
+                @SuppressWarnings("resource")
+                InetSocketAddress transportAddress = entry.getValue().localOpenSearchCluster.masterNode().getTransportAddress();
+                nodeOverride = Settings.builder().put(nodeOverride)
+                        .putList("cluster.remote." + entry.getKey() + ".seeds", transportAddress.getHostString() + ":" + transportAddress.getPort())
+                        .build();
+            }
+
+            start();
+        }
+    }
+
+    @Override
+    protected void after() {
+        if (localOpenSearchCluster != null && localOpenSearchCluster.isStarted()) {
+            try {
+                Thread.sleep(1234);
```

Comment:
This will probably give you a couple of error messages from the cluster, as there might be still some async things going on which need some time to settle. However, we can try to shorten that delay.

Reply:
Done, thread sleep removed.